File: apps/home-service-platform-server/src/APIs/Review/Dtos/ReviewWhereInput.cs
namespace HomeServicePlatform.APIs.Dtos;

public class ReviewWhereInput
{
    public string? Comment { get; set; }

    public DateTime? CreatedAt { get; set; }

    public string? Id { get; set; }

    public int? Rating { get; set; }

    public string? ServiceProvider { get; set; }

    public List<string>? ServiceProviders { get; set; }

    public DateTime? UpdatedAt { get; set; }

    public string? User { get; set; }

    public List<string>? Users { get; set; }
}

File: apps/home-service-platform-node-js-admin/src/api/review/ReviewWhereInput.ts
import { AdminListRelationFilter } from "../admin/AdminListRelationFilter";
import { StringNullableFilter } from "../../util/StringNullableFilter";
import { StringFilter } from "../../util/StringFilter";
import { IntNullableFilter } from "../../util/IntNullableFilter";
import { ServiceProviderWhereUniqueInput } from "../serviceProvider/ServiceProviderWhereUniqueInput";
import { ServiceProviderListRelationFilter } from "../serviceProvider/ServiceProviderListRelationFilter";
import { UserWhereUniqueInput } from "../user/UserWhereUniqueInput";

export type ReviewWhereInput = {
  admins?: AdminListRelationFilter;
  comment?: StringNullableFilter;
  id?: StringFilter;
  rating?: IntNullableFilter;
  serviceProvider?: ServiceProviderWhereUniqueInput;
  serviceProviders?: ServiceProviderListRelationFilter;
  user?: UserWhereUniqueInput;
};
